File: api/OrderMS.Application/Dtos/Responses/ApiResponse.cs


namespace OrderMS.Application.Dtos.Responses;

public record ApiResponse<T>
{
    public T? Data { get; set; }
    public string? Message { get; set; }
    public List<string> Errors { get; set; } = [];
    public bool Success { get; set; } = false;
}

File: ui/types/apiResponse.ts
export interface ApiResponse<T = any> {
  data: T | null;
  message?: string;
  errors?: string[];
  success?: boolean;
}

export interface PaginatedResult<T = any> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}